AzFileDiagnostics/Windows/AzFileDiagnostics.ps1

```diff
@@ -1427,6 +1427,11 @@ function GetServerProtocolSettings($PSStorageAccount) {
 
     $smbProtocolSettings = Get-AzStorageFileServiceProperty -StorageAccount $PSStorageAccount
 
+    if($null -eq $smbProtocolSettings.ProtocolSettings) {
```

Comment:
Is there a chance that ProtocolSettings is non-null, but ProtocolSettings.Smb is null?

Reply:
yes. if nobody every changed anything in protocol setting page of azure portal, we get it as NULL.

Reply:
Ok. Please also check if you're handling the same in Linux script too

Reply:
Yes. It being handled in Linux script.